### PR TITLE
Refactoring to es6

### PR DESCRIPTION
**Pull request by lpeterson52**
Refactored 4 modules (rotate, spur, walking, up-down) to es6 and got rid of dead variables in cam.js. Consolidated gear functions and got rid of useless for loops.

### CODE REVIEW

File: modules/js/cam.js [hunk 1]
```diff
@@ -212,9 +212,6 @@ Events.on(engine, "afterUpdate", function (event) {
   if (openCloseMod) {
     const bottom = compositeArray[0].constraints[0].pointA.y - rectBase;
     const top = compositeArray[0].bodies[0].position.y - 200 - pivotValue;
-    const pivotSpace =
-      compositeArray[0].constraints[0].pointA.y - 200 - pivotValue - bottom;
-    const rectWidth = compositeArray[2].width;
```
Comment (mklewandowski) [inline modules/js/cam.js:217]: I also like the addition of line breaks like this.  This is a good linting addition and makes the code more readable.

File: modules/js/rotate.js [hunk 2]
```diff
@@ -2,123 +2,158 @@
...
       Composite.remove(compositeArray[index], compositeArray[index].bodies[1]);
     }
     // remove gear body
     Composite.remove(compositeArray[index], compositeArray[index].bodies[0]);
     // store constraint position values
-    var tmpConstraintXPoint
-    if(index == 0){
-      tmpConstraintXPoint = (window.innerWidth)*(0.75*0.45)-(radius+(toothHeight*0.6))
+    let tmpConstraintXPoint;
+    if (index == 0) {
```
Comment (mklewandowski) [inline modules/js/rotate.js:15]: consider ===

File: modules/js/rotate.js [hunk 3]
```diff
@@ -2,123 +2,158 @@
...
+      tmpConstraintXPoint =
+        window.innerWidth * (0.75 * 0.45) - (radius + toothHeight * 0.6);
+    } else if (index == 1) {
```
Comment (mklewandowski) [inline modules/js/rotate.js:18]: Consider ===.  I won't comment on all of these equalities, but consider strict equality as part of moving to ES6:
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Strict_equality

File: modules/rotate.html [hunk 4]
```diff
@@ -0,0 +1,355 @@
...
+<!DOCTYPE html>
+<html>
+  <head>
+    <meta charset="UTF-8" />
+    <title>Rotating</title>
+    <script src="https://ajax.googleapis.com/ajax/libs/jquery/1.12.2/jquery.min.js"></script>
+    <link rel="shortcut icon" type="image/png" href="img/logo_webTab.png" />
+    <link
+      rel="stylesheet"
+      href="https://maxcdn.bootstrapcdn.com/bootstrap/3.3.6/css/bootstrap.min.css"
```
Comment (mklewandowski) [inline modules/rotate.html:22]: Another thing to look at for future work is updating the versions on these imports.  Bootstrap is on version 5 and version 3 came out over 10 years ago!

File: modules/js/rotate.js (hunk 3 above)
Comment (lpeterson52) [reply modules/js/rotate.js:18]: My bad, I'll go back through and change the operators

**Review by mklewandowski — APPROVED** [review]
This looks good!  I think all of my comments from the previous PR still apply but are outside the scope of this PR (things like module length and function names).  I would maybe look into using === (strict equality) in these conversions to ES6.  

Some other things that could be future work items/PRs:
- updating imports of external libraries like jQuery and Bootstrap. Although I think getting rid of jQuery would be nice since modern front-end has moved past it at this point.
- adding more comments to describe what some of the functions do.

But for now I vote to merge this!

**Pull request merged**